src/Service/Handler.php

```diff
@@ -180,15 +187,15 @@ public function handle(NodeInterface $node, string $action): void {
    *   The event data.
    */
   private function getEventData(NodeInterface $node) {
-    $valueHandler = new ValueHandler();
+    $valueHandler = new ValueHandler($this->fileUrlGenerator);
```

Comment:
This is weird, but it's not your fault. 

The `ValueHandler` should probably be dependency injected into this class rather than be created here (and every time `getEventData` is called). Furthermore, we should only create the `ObjectTransformer` using the value handler once, e.g. in the constructor where we get the value handler. Feel free to make this change now or postpone it to some later time.